pkg/testing/integration/program.go

```diff
@@ -1608,6 +1611,12 @@ func (pt *programTester) prepareGoProject(projinfo *engine.Projinfo) error {
 	if err != nil {
 		return err
 	}
+
+	// skip building if the 'go run' invocation path is requested.
+	if pt.opts.GoRunInvoke {
```

Comment:
We have a `RunBuild` flag already - make sense to reuse that here?

Reply:
I missed that, thanks for pointing it out. With this change, the default behavior will be for go programs to run without an explicit build step. I've updated the existing go integration test cases to maintain their current behavior, and to ensure we continue to have coverage on that path.